src/libs/blueprint/conduit_blueprint_mesh_examples.cpp

```diff
@@ -2532,6 +2532,175 @@ void spiral(index_t ndoms,
+    float64_array y_coords = res["coordsets/coords/values/y"].value(); 
+    
+    // 0 <-> 1
+    float64 dx = 1.0/float64(nx+1);
+    float64 dy = 1.0/float64(ny+1);
+
+    float64 vx = 0;
+    for(index_t i =0; i< nx+1; i++)
+    {
+        x_coords[i] = vx;
+        vx+=dx;
+    }
+    
+    float64 vy = 0;
+    for(index_t i =0; i< ny+1; i++)
+    {
+        y_coords[i] = vy;
+        vy+=dy;
+    }
+    
+    // create the topology
+    
+    res["topologies/topo/type"] = "rectilinear";
+    res["topologies/topo/coordset"] = "coords";
+
+    // create the fields
+
+    // circle a distance field
+    res["fields/radius_a/association"] = "element";
+    res["fields/radius_a/topology"] = "topo";
+    res["fields/radius_a/values"] = DataType::float64(nx * ny);
+    // circle a vf
+    res["fields/circle_a/association"] = "element";
+    res["fields/circle_a/topology"] = "topo";
+    res["fields/circle_a/values"] = DataType::float64(nx * ny);
+
+    // circle b distance field
+    res["fields/radius_b/association"] = "element";
+    res["fields/radius_b/topology"] = "topo";
+    res["fields/radius_b/values"] = DataType::float64(nx * ny);
+    // circle b vf
+    res["fields/circle_b/association"] = "element";
+    res["fields/circle_b/topology"] = "topo";
+    res["fields/circle_b/values"] = DataType::float64(nx * ny);
+
+    // circle c distance field
+    res["fields/radius_c/association"] = "element";
+    res["fields/radius_c/topology"] = "topo";
+    res["fields/radius_c/values"] = DataType::float64(nx * ny);
+    // circle b vf
+    res["fields/circle_c/association"] = "element";
+    res["fields/circle_c/topology"] = "topo";
+    res["fields/circle_c/values"] = DataType::float64(nx * ny);
+
+    // per element how many circles overlap
+    res["fields/overlap/association"] = "element";
+    res["fields/overlap/topology"] = "topo";
+    res["fields/overlap/values"] = DataType::float64(nx * ny);
+
+
+    float64_array rad_a = res["fields/radius_a/values"].value();
+    float64_array cir_a = res["fields/circle_a/values"].value();
+    
+    float64_array rad_b = res["fields/radius_b/values"].value();
+    float64_array cir_b = res["fields/circle_b/values"].value();
+
+    float64_array rad_c = res["fields/radius_c/values"].value();
+    float64_array cir_c = res["fields/circle_c/values"].value();
+
+
+    float64_array olap  = res["fields/overlap/values"].value();
+        
+    // circle a
+    // centered at:
+    //   x = 2/3 of width
+    //   y = 2/3 of width
+    
+    float64 a_center_x = 0.3333333333;
+    float64 a_center_y = 0.6666666666;
+
+    // circle b
+    // centered at:
+    //   x = 2/3 of width
+    //   y = 2/3 of width
+
+    float64 b_center_x = 0.6666666666;
+    float64 b_center_y = 0.6666666666;
+
+    // circle c
+    // centered at:
+    //   x = 1/2 of width
+    //   y = 2/3 of width
+    float64 c_center_x = 0.5;
+    float64 c_center_y = 0.3333333333;
+
+
+    float64 y = 0;
+    
+    index_t idx = 0;
```

Comment:
For index iteration situations like these, I've found embedding the iteration directly into the associated nested for-loop is useful for scoping and signalling intent, e.g.

```
for(index_t j = 0, idx = 0; j < ny; j++)
{
    for(index_t i = 0; i < nx; i++, idx++)
    {
        // ... implementation ...
    }
}
```
Given that the benefits of using this pattern are marginal and potentially offset by the nonstandard syntax, I think that either approach is fine; I just wanted to offer an alternative just in case it seemed preferable.